src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -420,14 +429,16 @@ private static boolean shouldCacheBids(ExtRequestTargeting targeting, ExtBidRequ
      * recorded response time.
      */
     private Future<BidderResponse> requestBids(BidderRequest bidderRequest, long startTime, Timeout timeout,
-                                               Map<String, String> aliases,
-                                               Map<String, BigDecimal> bidAdjustments) {
+                                               Map<String, String> aliases, Map<String, BigDecimal> bidAdjustments,
+                                               Currency currencyRates) {
         final String bidder = bidderRequest.getBidder();
         final BigDecimal bidPriceAdjustmentFactor = bidAdjustments.get(bidder);
+        // bidrequest.cur should always have only one currency in list, all other cases discarded by RequestValidator
+        final String adServerCurrency = bidderRequest.getBidRequest().getCur().get(0);
         return bidderCatalog.bidderRequesterByName(resolveBidder(bidder, aliases))
                 .requestBids(bidderRequest.getBidRequest(), timeout)
                 .map(this::validateAndUpdateResponse)
-                .map(seat -> applyBidPriceAdjustment(seat, bidPriceAdjustmentFactor))
+                .map(seat -> applyBidPriceChanges(seat, currencyRates, adServerCurrency, bidPriceAdjustmentFactor))
```

Comment:
What if to keep bid price adjustment and currency conversion as a separate stream operations?